release/version.py

```diff
@@ -0,0 +1,99 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+from typing import Optional
+import re
+import subprocess
+
+GIT_VERSION_FILE = "GIT_VERSION"
+
+
+def git() -> str:
+    """get the version from git. Can be
+    - X.Y.Z if a tag is set on the current HEAD
+    - X.Y.Z-<count>-g<commitid> otherwise where
+        - X.Y.Z is the latest version tag found
+        - <count> is the number of commits since then
+        - <commitid> is the current HEAD commitid
+    """
+    cmd = "git describe --tags --match=[0-9]*.[0-9]*.[0-9]*"
+    try:
+        p = subprocess.run(cmd.split(" "), capture_output=True, check=True)
+        v = p.stdout.decode("utf-8").strip()
+        # tox is running setup.py sdist from the git repo, and then runs again outside
+        # of the git repo. At this second step, we cannot run git commands.
+        # So we save the git version in a file and include it in the source distribution
+        with open(GIT_VERSION_FILE, "w") as f:
+            f.write(v)
+        return v
+    except subprocess.CalledProcessError as e:
+        # if we're not in a git repo, try reading out from the GIT_VERSION file
+        if os.path.exists(GIT_VERSION_FILE):
+            with open(GIT_VERSION_FILE, "r") as f:
+                return f.read()
+        print(e.stderr)
+        raise e
+
+
+def commit(length=7) -> str:
```

Comment:
Is this missing a type for parameter length? Or is this valid for the type checker? Not 100% sure

Reply:
can add ..